scim2-sdk-common/src/main/java/com/unboundid/scim2/common/messages/PatchOperation.java

```diff
@@ -48,17 +48,22 @@
 
 /**
  * An individual patch operation.
+ *
+ * Extended by collaboration Factory AG:
+ * - the @JsonSubTypes.Type also accept upper case strings for the operation types.
+ * - added a {@link PatchOperation#getOp} method in addition to {@link PatchOperation#getOpType()} in order to be compliant with the protocol
+ * - added a {@link PatchOperation#getPathString()} method that returns the path as String and replaces the {@link PatchOperation#getPath()} method for returning the path property
  */
 @JsonTypeInfo(use = JsonTypeInfo.Id.NAME,
     include = JsonTypeInfo.As.PROPERTY,
     property = "op")
 @JsonSubTypes({
     @JsonSubTypes.Type(value = PatchOperation.AddOperation.class,
-        name="add"),
+        name="add", names = {"Add", "ADD"}),
     @JsonSubTypes.Type(value = PatchOperation.RemoveOperation.class,
-        name="remove"),
+        name="remove", names = {"Remove", "REMOVE"}),
     @JsonSubTypes.Type(value = PatchOperation.ReplaceOperation.class,
-        name="replace")})
+        name="replace", names = {"Replace", "REPLACE"})})
```

Comment:
There is an open issue for this in the origin repository: https://github.com/pingidentity/scim2/issues/152